include/zephyr/drivers/can.h

```diff
@@ -825,9 +826,7 @@ struct can_device_state {
  *
  * @return 0 on success, or a negative error code on error
  */
-__syscall int can_get_core_clock(const struct device *dev, uint32_t *rate);
-
-static inline int z_impl_can_get_core_clock(const struct device *dev, uint32_t *rate)
```

Comment:
this works here where both the syscall and the implementation are in the same header, how is this going to work for something like https://github.com/zephyrproject-rtos/zephyr/blob/4a30aab2aefab891ba4b47ab2eed761c2210c53a/include/zephyr/kernel.h#L282?

Reply:
This is not an _inline_ syscall, ie, z_impl_* inline below. A _dummy_ `K_SYSCALL` macro would be enough to have the same scheme, ie

```c
#define K_SYSCALL(ret, name, ...) static inline ret name(__VA_ARGS__)
```

Reply:
ok, and how is 

 `K_SYSCALL(int, k_thread_stack_free, k_thread_stack_t *stack);` 

better than plain and direct 

`__syscall int k_thread_stack_free(k_thread_stack_t *stack); `?